BookEcommerce.Services/CartService.cs

```diff
@@ -32,103 +32,87 @@ public async Task<CartResponse> AddCart(CartRequest req, Guid cusId)
-                    var createCartDetail = new CartDetail
-                    {
-                        ProductVariantId = req.ProductVariantId,
-                        Quantity = req.Quantity,
-                    };
-                    await cartDetailRepository.AddAsync(createCartDetail);
-                }
-                else 
-                {
-                    var findCartDetail = await cartDetailRepository.GetListCartDetailByCartId(findCart.CartId);
-                    foreach (var item in findCartDetail)
-                    {
-                        if (item.ProductVariantId.Equals(req.ProductVariantId))
+                        item.Quantity += req.Quantity;
+                        await _unitOfWork.CommitTransaction();
+                        return new CartResponse
                         {
-                            item.Quantity += req.Quantity;
-                            await _unitOfWork.CommitTransaction();
-                            return new CartResponse
-                            {
-                                IsSuccess = true,
-                                Message = "Increased the number of this product in the cart Success!"
-                            };
-                        }
+                            IsSuccess = true,
+                            Message = "Increased the number of this product in the cart Success!"
+                        };
                     }
-                    var createCartDetail = new CartDetail
-                    {
-                        ProductVariantId = req.ProductVariantId,
-                        Quantity = req.Quantity,
-                        CartId = findCart.CartId,
-                    };
-                    await cartDetailRepository.AddAsync(createCartDetail);
                 }
+                var createCartDetail = new CartDetail
+                {
+                    ProductVariantId = req.ProductVariantId,
+                    Quantity = req.Quantity,
+                    CartId = findCart.CartId,
+                };
+                await cartDetailRepository.AddAsync(createCartDetail);
                 await _unitOfWork.CommitTransaction();
                 return new CartResponse
                 {
                     IsSuccess = true,
                     Message = "Add Product To Cart Success!"
                 };
             }
-            catch (Exception e)
+            catch (InvalidOperationException)
+            {
+                return new CartResponse
+                {
+                    IsSuccess = false,
+                    Message = "Some propaties is valid !",
+                };
+            }
+            catch (Exception)
             {
                 return new CartResponse 
                 { 
                     IsSuccess = false,
-                    Message = e.Message
+                    Message = "System error, Add Product to Cart was fasle and please try again later! "
                 };
```

Comment:
any log on this to find what actually wrong

Reply:
I will LogError("System error and Exception was not found! ")

Reply:
2 things are required on a log: e.Message and e.StackTrace. We use it to fix bug